include/glow/Base/Tensor.h

```diff
@@ -658,6 +658,62 @@ void dumpImpl(const Tensor *T, llvm::raw_ostream &os,
 void dumpImpl(const Tensor *T, unsigned maxNumElem);
 void dumpImpl(const Tensor *T);
 
+template <class ElemTy> class Handle;
+
+template <class ElemTy, bool IsConst>
+class HandleIterator : public std::iterator<std::forward_iterator_tag, ElemTy> {
+  using HandleTy = typename std::conditional_t<IsConst, const Handle<ElemTy> *,
+                                               Handle<ElemTy> *>;
+  using ElemTyRef =
+      typename std::conditional_t<IsConst, const ElemTy &, ElemTy &>;
+
+  HandleTy handle_;
+  llvm::ArrayRef<size_t> sizes_;
+  size_t indices_[max_tensor_dimensions] = {0};
+
+  HandleIterator(HandleTy handle) : handle_(handle) { sizes_ = handle->dims(); }
+
+public:
+  static HandleIterator begin(HandleTy handle) {
+    return HandleIterator(handle);
+  }
+
+  static HandleIterator end(HandleTy handle) {
+    auto result = HandleIterator(handle);
+    result.indices_[0] = result.sizes_.size() ? result.sizes_[0] : 1;
+    return result;
+  }
+
+  HandleIterator &operator++() {
+    size_t i = sizes_.size() ? sizes_.size() - 1 : 0;
```

Comment:
Layover approach can be a problem, may be if iterator hits the end, ++ doesn't move it to the beginning?